cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java

```diff
@@ -898,27 +845,6 @@ public void searchMetadata(HttpRequest request, HttpResponder responder,
       types = ImmutableSet.copyOf(Iterables.transform(targets, STRING_TO_TARGET_TYPE));
     }
 
-
-    String sortBy = "name";
-    String sortOrder = "asc";
-    if (!sort.isEmpty()) {
```

Comment:
what is the reason the validation logic is being moved from here to the store? I think it makes more sense to validate user input right at the start and have our classes handle SortInfo, rather than passing possibly invalid strings around.